### ISSUE TITLE
Redefining `Result` causes issues

### ISSUE THREAD

**Issue by Timmmm** [greyblake/envconfig-rs#21]
[This page](https://doc.rust-lang.org/rust-by-example/error/multiple_error_types/boxing_errors.html) suggests it should be normal to redefine `Result` and also use `#derive[Foo]`, however if you do it with envconfig you get an error:

```
#[macro_use]
extern crate envconfig_derive;

use envconfig::Envconfig;

#[derive(Envconfig)]
pub struct Config {
	#[envconfig(from = "API_KEY", default = "")]
	pub api_key: String,

	#[envconfig(from = "API_SECRET", default = "")]
	pub api_secret: String,
}

type Result<T> = std::result::Result<T, Box<std::error::Error>>;
```

```
error[E0107]: wrong number of type arguments: expected 1, found 2
 --> src/main.rs:6:10
  |
6 | #[derive(Envconfig)]
  |          ^^^^^^^^^ unexpected type argument
```

Could this be because the envconfig source uses `Result` rather than `std::Result`? It seems like a bit of a scoping flaw for Rust if you ask me, but anyway it would be nice to fix it.

**Comment by greyblake** [comment]
Hi! Thanks for reporting!

It's quite interesting issue :)

You guess is right!
Here `Result` is used: https://github.com/greyblake/envconfig-rs/blob/master/envconfig_derive/src/lib.rs#L45
I fixed it with using `::std::result::Result` instead.

**Comment by greyblake** [comment]
The fix is released in version `0.5.1`.
Thank you! :)

**Comment by Timmmm** [comment]
Excellent, thanks!
